Core/Sources/Implementations/species.cpp

```diff
@@ -71,13 +71,36 @@ auto Hippocrates::Species::IsStagnant() const -> bool {
 
 auto Species::LetPopulationLive() -> void {
 	for (auto& organism : population) {
-		while (!organism.HasFinishedTask()) {
-			organism.Update();
-		}
+		UpdateOrganism(organism);
 	}
+
 	isSortedByFitness = false;
 }
 
+auto Species::UpdateOrganism(Organism& organism) -> void {
+	if (GetTrainingParameters().structure.allowRecurrentConnections) {
+		UpdateOrganismWithMemoryResets(organism);
+	} else {
+		LetOrganismFinishTask(organism);
+	}
+}
+
+auto Species::UpdateOrganismWithMemoryResets(Organism& organism) -> void {
+	auto numberOfTimesToReset = GetTrainingParameters().structure.memoryResetsBeforeTotalReset;
+
+	while (--numberOfTimesToReset > 0) {
```

Comment:
shouldn't this be `numberOfTimesToReset--`? If not, why?
`std::size_t memoryResetsBeforeTotalReset = 0;` in the parameters suggests that this will underflow when using the standard parameters